node/src/types.ts

```diff
@@ -33,22 +53,11 @@ export interface Document {
    */
   name: string
   /**
+   * The format of the document.
+   *
    * On initialization, this is inferred, if possible, from the file name extension of the document's `path`. However, it may change whilst the document is open in memory (e.g. if the `load` function sets a different format).
    */
-  format: {
-    /**
-     * The lowercase name of the format e.g. `md`, `docx`, `dockerfile`
-     */
-    name: string
-    /**
-     * Whether or not the format should be considered binary e.g. not to be displayed in a text / code editor
-     */
-    binary: boolean
-    /**
-     * The type of `CreativeWork` that this format is expected to be. This will be `None` for data serialization formats such as JSON or YAML which have no expected type (the actual type is embedded in the data).
-     */
-    type?: string
-  }
+  format: Format
```

Comment:
Nothing major, but want to make sure we're on the same page.
I thought we had decided on keeping the formats as a simple string, and having a lookup function mapping it to the various attributes?

Reply:
Yes, we decided to keep formats a simple string and do the lookup. But then we discussed that for the files browser in particular this would require extra expense to do the lookup for something that will change very infrequently. So I thought we reverted the first decision and decided to go with an object instead of a string.

Reply:
ah, pardon the confusion. I meant to say that there will be overhead, but nothing to concern ourselves about—meaning we should use the strings 🙃 
No worries either way

Reply:
No worries. It feels like using `Format` instead of `strings` is less brittle (i.e. less code, the lookup only gets done in one place) so lets stick with that.